Automattic/jetpack
# Social | Simplify and clarify social network selection for post sharing

Currently the messaging on the connection toggle section on post edit page is a bit confusing as highlighted here pdrWKz-1xe-p2

## Proposed changes:
* Update the connection related messaging to make the intent clear
* Simplify the logic used to render the messages

### Other information:


## Jetpack product discussion
pdrWKz-1xe-p2

## Does this pull request change what data or activity we track or use?



## Testing instructions:






Follow these instructions for both Jepack and Jetpack Social individually as well as when both are installed

- For a site with some social connections added, create a new post
- Open the Jetpack or Social sidebar
- Confirm that the connection messages says, `"Enable the social media accounts where you want to share your post."` by default.
- Turn OFF Publicize (Share when publishing)
- Confirm that the message changes to `"Use this tool to share your post on all your social media accounts."`
- Turn ON Publicize again
- Confirm that enabling or disabling connections does not change the message
- Open an already published post
- Open the Jetpack or Social sidebar
- Confirm that the message always says, `"Enable the social media accounts where you want to re-share your post, then click on the "Share post" button below."`

## Screenshots
When the post is already published, we show the same message, whether any of the connections is enabled or not
<img width="287" alt="Screenshot 2024-04-04 at 3 53 31 PM" src="https://github.com/Automattic/jetpack/assets/18226415/83b7855d-73fc-4180-a5e4-a1e7ce49ccd0">

For new posts, we show "Use this tool to share..." if publicize is not enabled
<img width="288" alt="Screenshot 2024-04-04 at 4 07 01 PM" src="https://github.com/Automattic/jetpack/assets/18226415/8996c1a6-6529-415e-bf87-606a5a5b111d">

Otherwise show the enabled connections message
<img width="286" alt="Screenshot 2024-04-04 at 4 07 22 PM" src="https://github.com/Automattic/jetpack/assets/18226415/348b61c3-0cd6-4c43-b4c7-5c91a7d9b07f">
